packages/components/src/modal/index.tsx

```diff
@@ -36,8 +42,13 @@ import Button from '../button';
 import StyleProvider from '../style-provider';
 import type { ModalProps } from './types';
 
-// Used to count the number of open modals.
-let openModalCount = 0;
+// Used to track and dismiss the prior modal when another opens unless nested.
+const level0Dismissers: MutableRefObject<
+	ModalProps[ 'onRequestClose' ] | undefined
+>[] = [];
+const context = createContext( level0Dismissers );
```

Comment:
Nit: let's capitalize the variable, and give it a more meaningful name (this better aligns with naming conventions and can help while debugging code in React Dev Tools)

Reply:
This and the inline comments are addressed in 145dc0e.